packages/nitro-rpc-client/scripts/client-runner.ts

```diff
@@ -134,3 +153,47 @@ async function wait(ms: number) {
 function getChannelIdFromObjectiveId(objectiveId: string): string {
   return objectiveId.split("-")[1];
 }
+
+// Waits for the RPC server to be available by sending a simple get_address POST request until we get a response
+async function waitForRPCServer(
+  port: number,
+  waitDuration: number
+): Promise<void> {
+  const startTime = Date.now();
+  while (Date.now() - startTime < waitDuration) {
+    const isUp = await isServerUp(port);
+    if (isUp) {
+      console.log(
+        `RPC server ${getLocalRPCUrl(port)} is responding!
+        Waited ${
+          Date.now() - startTime
+        } milliseconds for server to be responsive`
+      );
+
+      return;
+    } else {
+      console.log(`RPC server ${getLocalRPCUrl(port)} not available, waiting!`);
+      await wait(1000);
+    }
+  }
+  throw new Error(
+    `RPC server ${getLocalRPCUrl(port)} not reachable in ${waitDuration} ms`
+  );
+}
+
+// Checks if the server is up by sending a simple get_address POST request
+// This is specific to the HTTP/WS RPC transport
+async function isServerUp(port: number): Promise<boolean> {
+  let result: AxiosResponse<unknown, unknown>;
+
+  try {
+    const req = generateRequest("get_address", {});
+    result = await axios.post(
```

Comment:
Is it weird that this uses http POST rather than GET?

Reply:
Why not use the rpc client to launch the `get_address` request?

Reply:
> Is it weird that this uses http POST rather than GET?

Our `go-nitro` RPC server **only accepts** POST request, so we have to use POST here.

Reply:
> > Is it weird that this uses http POST rather than GET?
> 
> Our `go-nitro` RPC server **only accepts** POST request, so we have to use POST here.

My question really was, it is weird the server only accepts POST requests?

Reply:
> My question really was, it is weird the server only accepts POST requests?

I think it kinda makes sense. The JSON-RPC interface is accessed using one endpoint, and different methods are called by specifying the `method` in your JSON-RPC request data. Even for cases like `get_address` , that do not require any method  parameters,  the consumer is still sending  JSON data in the POST body.  For example in the case of `getAddress` we're posting JSON that looks like this:
```
'{"jsonrpc":"2.0","method":"getAddress","params":[],"id":67}'
```

Reply:
https://www.simple-is-better.org/json-rpc/transport_http.html#get-request